### PR TITLE
update setSlotAddress and unit test

### PR DESCRIPTION
**Pull request by hiutin0**

### CODE REVIEW

File: src/test/scala/vsys/state/diffs/SPOSTransactionDiffTest.scala [hunk 1]
```diff
@@ -65,6 +67,30 @@ class SPOSTransactionDiffTest extends PropSpec with PropertyChecks with Generato
...
+    master1 <- accountGen
+    ts <- positiveIntGen
+    genesis: GenesisTransaction = GenesisTransaction.create(master, ENOUGH_AMT, 0, ts).right.get
+    genesis1: GenesisTransaction = GenesisTransaction.create(master1, 2*ENOUGH_AMT, -1, ts).right.get
+    contend: ContendSlotsTransaction <- contendGeneratorP(master1, 0)
+    contend1: ContendSlotsTransaction <- contendGeneratorP(master, 1)
+  } yield (genesis, genesis1, contend, contend1, contend.fee)
+
+
+  property("contend transaction is successful") {
```
Comment (ncying) [inline src/test/scala/vsys/state/diffs/SPOSTransactionDiffTest.scala]: may change to re-contend transaction is successful~

**Comment by ncying** [comment]
unit test passed
```scala
[info] ScalaTest
[info] Run completed in 45 seconds, 951 milliseconds.
[info] Total number of tests run: 325
[info] Suites: completed 102, aborted 0
[info] Tests: succeeded 325, failed 0, canceled 0, ignored 15, pending 2
[info] All tests passed.
[info] Passed: Total 325, Failed 0, Errors 0, Passed 325, Ignored 15, Pending 2
[success] Total time: 110 s, completed Feb 25, 2019 2:27:33 PM
```

File: src/main/scala/com/wavesplatform/state2/StateStorage.scala [hunk 2]
```diff
@@ -31,6 +31,10 @@ class StateStorage private(file: Option[File]) extends AutoCloseable {
   val addressToID: MVMap[String, Int] = db.openMap("addressToID")
 
   def setSlotAddress(i: Int, add: String):Unit = {
+    if (getSlotAddress(i).isDefined) {
+      addressToID.remove(addressList.get(i))
+      addressList.remove(i)
```
Comment (augmentedeconomy) [inline src/main/scala/com/wavesplatform/state2/StateStorage.scala]: is this line necessary?

**Comment by ncying** [comment]
case: super-node who contended by others can not contend new slot

reason: `setSlotAddress` function kept the old `address -> slot id` map

however, it seems a good advantage that node can not re-contend a slot in a short time.
need to 
1. create a new address
2. collect new lease in
3. wait the new MAB reach the bound

result: keeping this property (by @sunnyking ) and update the document

@augmentedeconomy

**Comment by augmentedeconomy** [comment]
> case: super-node who contended by others can not contend new slot
> 
> reason: `setSlotAddress` function kept the old `address -> slot id` map
> 
> however, it seems a good advantage that node can not re-contend a slot in a short time.
> need to
> 
> 1. create a new address
> 2. collect new lease in
> 3. wait the new MAB reach the bound
> 
> result: keeping this property (by @sunnyking ) and update the document
> 
> @augmentedeconomy

I agree with you that it is good that a node has to wait for some time to re-contend. but is your comment related to my comment regarding "addressList.remove(i)"?

**Comment by ncying** [comment]
@augmentedeconomy

yes, you are right, it's a useless code. since the old map will be replaced by `put` . (@hiutin0 )

this pr will be closed later.

**Pull request closed without merging**